packages/framer-motion/src/gestures/use-tap-gesture.ts

```diff
@@ -70,36 +78,44 @@ export function useTapGesture({
         visualElement.getProps().onTapCancel?.(event, info)
     }
 
-    const startPress = useCallback(
-        (event: PointerEvent, info: EventInfo) => {
-            removePointerEndListener()
+    function onPointerStart(event: PointerEvent, info: EventInfo) {
+        const latestProps = visualElement.getProps()
 
-            if (isPressing.current) return
-            isPressing.current = true
+        /**
+         * Ensure we trigger animations before firing event callback
+         */
+        if (latestProps.whileTap && visualElement.animationState) {
+            visualElement.animationState.setActive(AnimationType.Tap, true)
+        }
 
-            cancelPointerEndListeners.current = pipe(
-                addPointerEvent(window, "pointerup", onPointerUp, eventOptions),
-                addPointerEvent(
-                    window,
-                    "pointercancel",
-                    onPointerCancel,
-                    eventOptions
-                )
-            )
+        latestProps.onTapStart?.(event, info)
+    }
 
-            const latestProps = visualElement.getProps()
+    const callbackDependencies = [
+        Boolean(onTapStart),
+        Boolean(onTap),
+        Boolean(whileTap),
+        visualElement,
```

Comment:
Why shouldn't we use the explicit tap handlers as dependencies here?

Reply:
Because we don't use these directly but via `visualElement.getProps()`. It's only their presence that informs whether we create a new event handler.

Reply:
If we used them directly they would probably change every render.